### ISSUE TITLE
setting photos must be approved before public is not working for profile and cover photo and show in feed

### ISSUE THREAD

**Issue by spirogg** [PHPfox-Official/phpfox-v4-issues#2947]
# Important

Please follow this template!

----------------------------------------

## What's happened?
...
  @harrison05 i made this into another issue: not sure because  2 different issues unless you consider this the same as this #2945 ?

- when a new user signs up for a new account and you have this setting (Photos must be approved first before they are displayed publicly?) set to YES 

- **User Profile photo should not be available publicly**
& 
- **Cover photo should not be available publicly** 

_**They Should be pending approval from admin like all photos uploaded.**_

## Steps to reproduce:
1. in admincp manage usergroups
2. find registered users and select photos
3. find (Photos must be approved first before they are displayed publicly?) set to YES 
4. singup as new user and upload profile photo and cover photo - all users can see publicly and also the main feed will display both these uploaded photos.   
## What's expected?
...should not show these photos and should be pending for approval from admin, because spammer or idiots can upload sex photos or something not good for other users of website.

## Browsers and Devices tested
(Example: Chrome on iPhone X, Safari on Macbook, Miscrosoft Edge on Windown 10, Firefox on Ubuntu 16.04, ...)

... all browsers 

## Server information
(Example: CentOS 7, php 7.1 apache)

...CentOS 7 nginx php 7.4

## phpFox version
(Example: phpFox 4.6.0)
4.8.1 fresh install
...

## Screenshots
...
![Capture7](https://user-images.githubusercontent.com/17736688/96823554-5ab8b580-13f2-11eb-856f-385480284ac8.PNG)
![Capture8](https://user-images.githubusercontent.com/17736688/96823555-5ab8b580-13f2-11eb-9655-fba58300475a.PNG)
![Capture6](https://user-images.githubusercontent.com/17736688/96823594-6e641c00-13f2-11eb-8c1d-18d8d62bb722.PNG)
![Capture9](https://user-images.githubusercontent.com/17736688/96823595-6e641c00-13f2-11eb-888e-20a489e613e4.PNG)

**Comment by spirogg** [comment]
lets say I am a spammer and want to post sex or obscene photographs they can do this in profile and cover photo and all users on feed can see this?  trying to eliminate this for that reason. but if they delete the profile photo or cover photo or any photo it should go back to needs approval so we can catch to spammers photos and decline them? 
not sure if this is possible ?

**Comment by harrison05** [comment]
Hi @spirogg 

We will check and improve the workflow of this case. 
Solution: When users upload avatar/cover but setting "Photos must be approved first before they are displayed publicly?" is enabled, their photos will be marked as pending even though they can see them on their profile page but don't have feeds and other can't view photos detail until an admin approves them.

Regards.

**Comment by spirogg** [comment]
> Hi @spirogg
> 
> We will check and improve the workflow of this case.
> Solution: When users upload avatar/cover but setting "Photos must be approved first before they are displayed publicly?" is enabled, their photos will be marked as pending even though they can see them on their profile page but don't have feeds and other can't view photos detail until an admin approves them.
> 
> Regards.

HELLO @harrison05 

- I update to version 4.8.2 and now when user try to upload Avatar / Cover photo, it asks for approval :)

-  **But the avatar / cover does not change in profile section or cover section?**  
- after the admin reviews and approves, the avatar photo goes to profile photos section only and does not change the actual profile avatar / and the cover photo for that user is not updated and just shows as a photo in photo section but not cover photo?


- **ALSO PLEASE NOTE**
( after admin approves Avatar / Cover photo for user, they do not show in the Main Feed as updated profile or updated cover for that user? )

- If I update my Cover / Avatar it will show: ( **Admin updated their profile photo** and **Admin updated their cover photo** in main feed )

**it should say this in Main feed as well please, after we approved the Avatar / Cover .**


-
-
-

- see images below of my tests

- 
![Captures](https://user-images.githubusercontent.com/17736688/98481420-b4292e80-21bf-11eb-9eca-b03167eac376.PNG)
![Capture](https://user-images.githubusercontent.com/17736688/98481414-b25f6b00-21bf-11eb-87c5-f5fefc963330.PNG)
![Capture1](https://user-images.githubusercontent.com/17736688/98481415-b2f80180-21bf-11eb-87ac-dc5c20986e25.PNG)
![Capture2](https://user-images.githubusercontent.com/17736688/98481416-b2f80180-21bf-11eb-814e-d2aea38bfc64.PNG)
![Capture3](https://user-images.githubusercontent.com/17736688/98481417-b3909800-21bf-11eb-8b45-97430cff734b.PNG)
![Capture4](https://user-images.githubusercontent.com/17736688/98481418-b3909800-21bf-11eb-8552-d34d6c60c08d.PNG)
![Capture5](https://user-images.githubusercontent.com/17736688/98481419-b3909800-21bf-11eb-937e-027e25042d36.PNG)



- @harrison05 this is for cover photo showing its not working 

![image](https://user-images.githubusercontent.com/17736688/98481577-cd7eaa80-21c0-11eb-9bba-0d01fde3b25a.png)

**Comment by spirogg** [comment]
@harrison05 Same thing happens with user cover photo, it does not change for the user even after the admin approves the cover photo ?

- **ALSO PLEASE NOTE**
( after admin approves profile photo oe cover photo for user, they do not show in the Main Feed as updated profile or updated cover for that user ?)

![image](https://user-images.githubusercontent.com/17736688/98481577-cd7eaa80-21c0-11eb-9bba-0d01fde3b25a.png)

**Comment by harrison05** [comment]
Hi @spirogg 

This feature improvement required the new version of Photo App v4.7.11. We will release it soon.

Regards.

**Comment by kibcode** [comment]
@harrison05 seems like v4.7.11 broke the process. wenn users are forced to upload a profile photo on registration and the setting to approve first is disabled all the photos are pending.

**Comment by spirogg** [comment]
@harrison05 my initial issue is fixed with phpfox 4.8.2 and Photo App v4.7.11 its working..

@harrison05 
@Scheinwelt-Media has another issue you can please check..  

- so far your fixes are working great thank you..

**Comment by spirogg** [comment]
> @harrison05 seems like v4.7.11 broke the process. wenn users are forced to upload a profile photo on registration and the setting to approve first is disabled all the photos are pending.

**I can confirm** 

- when user signs up if not required to approve photos but forced to upload an avatar- then the initial photo (avatar) uploaded is pending.. 

but even though its initial avatar is pending, the user can still upload another avatar and it will work.

- so we need fix the initial upload in this case.. nice find @Scheinwelt-Media

**Comment by spirogg** [comment]
PS the feature: `users are forced to upload a profile photo on registration `is a little redundant 
when we have a user group setting: `Users are required to upload a profile image?` 

**seems this should be one or the other**  this is my opinion.. 
- because if you don't force at registration you can force when they login to upload a avatar.

- see photo below.
- 
- ![Capture](https://user-images.githubusercontent.com/17736688/98801792-f2993600-23d7-11eb-8e46-b3090bbfaa76.PNG)
-
- so then user logs in and can only see upload photo before he enters the website
-
- ![Capture](https://user-images.githubusercontent.com/17736688/98802596-26289000-23d9-11eb-9545-14d21b3cd768.PNG)
-
- then user uploads photo and it works does not get stuck in pending photos this way
- 
- ![Capture1](https://user-images.githubusercontent.com/17736688/98802701-49ebd600-23d9-11eb-9709-fa761d9b5d20.PNG)

**Comment by harrison05** [comment]
Hi all,

This might a bug, we will check and fix it.

Thanks
